src/Raven.Server/Documents/SchemaValidation/Validators/Untyped/FixedValueSchemaRuleValidator.cs

```diff
@@ -5,23 +5,40 @@ namespace Raven.Server.Documents.SchemaValidation.Validators.Untyped;
 
 public abstract class FixedValueSchemaRuleValidator : SchemaRuleValidator<object>
 {
-    protected static object ConvertTypeForComparison(object v)
+    protected static object ConvertTypeForComparison(object v, bool cloneAsRoot)
     {
         return v switch
         {
             LazyNumberValue lnv => (decimal)lnv,
             long lv => (decimal)lv,
-            decimal 
-                or LazyStringValue 
-                or LazyCompressedStringValue 
-                or BlittableJsonReaderObject 
-                or BlittableJsonReaderArray 
-                or bool 
-                or null => v,
+            LazyStringValue or LazyCompressedStringValue => v.ToString(),
```

Comment:
Can you avoid materializing to a string?
Save the pointer, size and the type and compare it based on that.